From ca0d44eadabefa305c30bd5af43a33a852fc4a6d Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 16 Nov 2021 14:25:13 +0100
Subject: [PATCH] setup-disk: fix kernel options with encrypted root

We should append the kernel options for cryptroot, not replace. This is
so nomodeset, quiet and other options are kept.
---
 setup-disk.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-disk.in b/setup-disk.in
index 5754c04..2f310fc 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -619,7 +619,7 @@ install_mounted_root() {
 		if cryptsetup status "$cryptroot" 2>&1 >/dev/null; then
 			cryptroot=$(cryptsetup status "$cryptroot" | awk '/device:/ { print $2 }')
 			cryptroot=$(uuid_or_device $cryptroot)
-			kernel_opts="cryptroot=$cryptroot cryptdm=root"
+			kernel_opts="cryptroot=$cryptroot cryptdm=root $kernel_opts"
 			root=$([ -n "$pvs" ] && echo "$rootdev" || echo "/dev/mapper/root")
 		fi
 
-- 
2.34.0

